R/generateIIASASubmission.R

```diff
@@ -26,49 +31,93 @@
+#'        If `outputDirectory` is set to NULL, this parameter has no effect.
 #' @param outputFilename filename of the generated submission. Must be mif or xlsx file.
+#'        If NULL, submission data is returned.
+#'        If `outputDirectory` is set to NULL, this parameter has no effect.
 #' @param iiasatemplate optional filename of xlsx or yaml file provided by IIASA
 #'        used to delete superfluous variables and adapt units
 #' @param generatePlots boolean, whether to generate plots of failing summation checks
-#' @param generateSingleOutput has no effect and is only kept for backwards-compatibility
 #' @param timesteps timesteps that are accepted in final submission
-#' @importFrom data.table :=
+#' @param mappingFile has no effect and is only kept for backwards-compatibility
 #' @importFrom quitte as.quitte write.IAMCxlsx write.mif
 #' @importFrom dplyr filter mutate distinct inner_join
-#' @importFrom magclass unitsplit
 #' @importFrom stringr str_trim
 #' @examples
 #' \dontrun{
 #' # Simple use. Generates submission file in output folder:
 #' generateIIASASubmission(
 #'   mifs = "/path/to/REMIMD/mifs",
 #'   model = "REMIND-MAgPIE 2.1-4.2",
-#'   mappingFile = "output/template_navigate.csv"
+#'   mapping = "NAVIGATE"
 #' )
 #' }
 #' @export
-generateIIASASubmission <- function(mifs = ".", mapping = NULL, model = NULL, # nolint cyclocomp_linter
-                                    mappingFile = NULL,
-                                    removeFromScen = NULL, addToScen = NULL,
+generateIIASASubmission <- function(mifs = ".", # nolint cyclocomp_linter
+                                    mapping = NULL,
+                                    model = NULL,
+                                    removeFromScen = NULL,
+                                    addToScen = NULL,
                                     outputDirectory = "output",
-                                    logFile = paste0(gsub("\\.[a-zA-Z]+$", "_log.txt", outputFilename)),
                                     outputFilename = "submission.xlsx",
-                                    iiasatemplate = NULL, generatePlots = FALSE,
+                                    logFile = NULL,
+                                    iiasatemplate = NULL,
+                                    generatePlots = FALSE,
                                     timesteps = c(seq(2005, 2060, 5), seq(2070, 2100, 10)),
-                                    generateSingleOutput = TRUE) {
+                                    mappingFile = NULL) {
+
+  # process input parameters ----
+
+  if (! is.null(mappingFile)) {
+    warning("mappingFile is deprecated and ignored. If you got here via output.R -> export -> xlsx_IIASA,
+            please pick a newer xlsx_IIASA.R file from remindmodel/develop")
+  }
 
   if (isTRUE(timesteps == "all")) timesteps <- seq(1, 3000)
+
+  # if output directory is not set, do not write a log file or output file
+  if (is.null(outputDirectory)) {
+    logFile <- FALSE
```

Comment:
Please remove the logFile stuff here. Because if you set outputDirectory to NULL, you still might want to pass `logFile = NULL` (different to FALSE) in order for the unit checks etc. to report to stdout.

Reply:
As of now, it would crash otherwise. I recommend, you fix this in a separate PR that unifies handling of the logFile parameter

Reply:
Sorry, you were right. Setting it to NULL works. Has been adjusted